src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpJupyterDAO.scala

```diff
@@ -0,0 +1,31 @@
+package org.broadinstitute.dsde.workbench.leonardo.dao
+
+import org.broadinstitute.dsde.workbench.leonardo.dns.ClusterDnsCache._
+import akka.actor.{ActorRef, ActorSystem}
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model._
+import akka.pattern.ask
+import akka.stream.ActorMaterializer
+import akka.util.Timeout
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.workbench.leonardo.model.google.ClusterName
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+import scala.concurrent.duration._
+import scala.concurrent.{ExecutionContext, Future}
+
+class HttpJupyterDAO(val clusterDnsCache: ActorRef)(implicit system: ActorSystem, materializer: ActorMaterializer, executionContext: ExecutionContext) extends JupyterDAO with LazyLogging {
+
+  val http = Http(system)
+
+  override def getStatus(googleProject: GoogleProject, clusterName: ClusterName): Future[Boolean] = {
+    getTargetHost(googleProject, clusterName) flatMap {
+      case ClusterReady(targetHost) => http.singleRequest(HttpRequest(uri = Uri(s"https://${targetHost.toString}/notebooks/$googleProject/$clusterName/api/status"))) flatMap(response => Future.successful(response.status.isSuccess()))
```

Comment:
Sorry another nitpick. :) I don't think you need to `flatMap` the http response since it's always returning `Future.successful` -- you can just `map` it and return the boolean directly.

Also the line length is a little long (I'm channelling Kaan here). 

Maybe:
```
val statusUri = Uri(s"https://${targetHost.toString}/notebooks/$googleProject/$clusterName/api/status")
http.singleRequest(HttpRequest(uri = statusUri)) map { response =>
  response.status.isSuccess
}
```